packages/tools/client-debug-view/src/components/data-object-views/SharedCounterView.tsx

```diff
@@ -37,12 +54,65 @@ export function SharedCounterView(props: SharedCounterViewProps): React.ReactEle
 		};
 	}, [sharedCounter, setValue]);
 
+
+    function decrementCounter(): void {
+        sharedCounter.increment(-deltaValue);
+	}
+
+    function incrementCounter(): void {
+		sharedCounter.increment(deltaValue);
+	}
+
+    const inputSetValue = (e: React.ChangeEvent<HTMLInputElement>): void => {
+        numberFormatCheck(e.target.value);
+    }
+
+    const numberFormatCheck = (val: string): void => {
+        const num = Number.parseInt(val, 10) || value;
+
+        if (!Number.isFinite(num) || num[0] === '-') {
```

Comment:
Just curious, when does the event give us "-"? Might be worth adding a brief comment about that if we expect it.